fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -639,106 +640,111 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
+                    if (log.isLoggable(Level.FINE)) {
+                        log.fine("searchResultCount = " + searchResultCount);
                     }
+                    searchContext.setTotalCount(searchResultCount);
                 }
+            }
 
-                // For _summary=count or pageSize == 0, we return only the count
-                if (searchResultCount > 0
-                        && !SummaryValueSet.COUNT.equals(searchContext.getSummaryParameter())
-                        && searchContext.getPageSize() > 0) {
-                    query = queryBuilder.buildQuery(resourceType, searchContext);
-
-                    List<String> elements = searchContext.getElementsParameters();
-
-                    //Only consider _summary if _elements parameter is empty
-                    if (elements == null && searchContext.hasSummaryParameter()) {
-                        Set<String> summaryElements = null;
-                        SummaryValueSet summary = searchContext.getSummaryParameter();
-
-                        switch (summary) {
-                        case TRUE:
-                            summaryElements = JsonSupport.getSummaryElementNames(resourceType);
-                            break;
-                        case TEXT:
-                            summaryElements = SearchUtil.getSummaryTextElementNames(resourceType);
-                            break;
-                        case DATA:
-                            summaryElements = JsonSupport.getSummaryDataElementNames(resourceType);
-                            break;
-                        default:
-                            break;
-                        }
+            List<OperationOutcome.Issue> issues = validatePagingContext(searchContext);
+            if (!issues.isEmpty()) {
+                resultBuilder.outcome(OperationOutcome.builder()
+                    .issue(issues)
+                    .build());
+                if (!searchContext.isLenient()) {
+                    return resultBuilder.success(false).build();
+                }
+            }
 
-                        if (summaryElements != null) {
-                            elements = new ArrayList<>();
-                            elements.addAll(summaryElements);
-                        }
+            // For _summary=count or pageSize == 0, we return only the count
+            if ((searchResultCount == null || searchResultCount > 0)
+                    && !SummaryValueSet.COUNT.equals(searchContext.getSummaryParameter())
+                    && searchContext.getPageSize() > 0) {
+                query = queryBuilder.buildQuery(resourceType, searchContext);
+
+                List<String> elements = searchContext.getElementsParameters();
+
+                // Only consider _summary if _elements parameter is empty
+                if (elements == null && searchContext.hasSummaryParameter()) {
+                    Set<String> summaryElements = null;
```

Comment:
Minor, but you could make summaryElements final, and then set to null in the default of the switch. It's useful to avoid missed break statements which can end up assigning the value value more than once.

Reply:
This isn't code I changed, but rather whitespace-only changes due to indenting. I'd prefer to just leave it as is.